src/textual/app.py

```diff
@@ -931,21 +931,20 @@ def bell(self) -> None:
         """Play the console 'bell'."""
         self.console.bell()
 
-    async def press(self, key: str) -> bool:
+    async def press(self, event: events.Key) -> bool:
```

Comment:
the `action` method now needs to be able to optionally receive the event, so we can process the `$event.[something]` placeholders of the binding expression